transaction/tests/Lock_unittest.cpp

```diff
@@ -0,0 +1,102 @@
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "transaction/Lock.hpp"
+
+#include <cstddef>
+#include <vector>
+
+#include "transaction/AccessMode.hpp"
+#include "transaction/ResourceId.hpp"
+
+#include "gtest/gtest.h"
+
+namespace quickstep {
+namespace transaction {
+
+class LockTest : public ::testing::Test {
+ protected:
+  LockTest()
+      : modes_({AccessMode(AccessModeType::kNoLock),
+                AccessMode(AccessModeType::kIsLock),
+                AccessMode(AccessModeType::kIxLock),
+                AccessMode(AccessModeType::kSLock),
+                AccessMode(AccessModeType::kSixLock),
+                AccessMode(AccessModeType::kXLock)}),
+        resource_a_(3, 10, 2, 5),
+        resource_b_(4, 5, 3, 2),
+        resource_a_copy_(resource_a_),
+        locks_on_resource_a_({Lock(resource_a_, modes_[0]),
+                              Lock(resource_a_, modes_[1]),
+                              Lock(resource_a_, modes_[2]),
+                              Lock(resource_a_, modes_[3]),
+                              Lock(resource_a_, modes_[4]),
+                              Lock(resource_a_, modes_[5])}),
+        locks_on_resource_b_({Lock(resource_b_, modes_[0]),
+                             Lock(resource_b_, modes_[1]),
+                             Lock(resource_b_, modes_[2]),
+                             Lock(resource_b_, modes_[3]),
+                             Lock(resource_b_, modes_[4]),
+                             Lock(resource_b_, modes_[5])}),
+       locks_on_resource_a_copy_({Lock(resource_a_copy_, modes_[0]),
+                                  Lock(resource_a_copy_, modes_[1]),
+                                  Lock(resource_a_copy_, modes_[2]),
+                                  Lock(resource_a_copy_, modes_[3]),
+                                  Lock(resource_a_copy_, modes_[4]),
+                                  Lock(resource_a_copy_, modes_[5])}) {
+  }
+
+  const std::vector<AccessMode> modes_;
+  const ResourceId resource_a_;
+  const ResourceId resource_b_;
+  const ResourceId resource_a_copy_;
+  const std::vector<Lock> locks_on_resource_a_;
+  const std::vector<Lock> locks_on_resource_b_;
+  const std::vector<Lock> locks_on_resource_a_copy_;
+};
+
+TEST_F(LockTest, LockEquality) {
+  // Locks are equal if they are on the same resource with same
+  // access mode.
+  for (std::size_t i = 0; i < locks_on_resource_a_.size(); ++i) {
+    for (std::size_t j = 0; j < locks_on_resource_a_copy_.size(); ++j) {
```

Comment:
You don't actually a copy, use `locks_on_resource_a_` to replace `locks_on_resource_a_copy_`, and remove `locks_on_resource_a_copy_` from `LockTest`.